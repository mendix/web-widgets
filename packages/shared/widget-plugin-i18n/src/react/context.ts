import { createContext } from "react";
import { PluginI18n } from "../types";

const contextPath = "com.mendix.widget.plugin.i18n.context" as const;

type I18nContext = React.Context<PluginI18n | null>;

const createI18nContext = (): I18nContext => createContext<PluginI18n | null>(null);

declare global {
    interface Window {
        [contextPath]?: ReturnType<typeof createI18nContext>;
    }
}

export function getGlobalI18nPluginContext(): I18nContext {
    return window[contextPath] ?? createI18nContext();
}
