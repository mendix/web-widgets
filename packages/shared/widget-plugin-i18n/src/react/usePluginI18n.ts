import { useState } from "react";
import { I18nController } from "../controllers/I18nController";
import { PluginI18n } from "../types";

export function usePluginI18n(languageCode: string): PluginI18n | null {
    const [plugin, setPlugin] = useState<PluginI18n | null>(() => {
        new I18nController({
            languageCode,
            onPluginChange: plugin => setPlugin(plugin)
        });

        return null;
    });

    return plugin;
}
