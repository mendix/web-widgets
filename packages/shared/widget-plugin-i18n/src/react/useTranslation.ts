import { useContext } from "react";
import { PluginI18n } from "../types";
import { getGlobalI18nPluginContext } from "./context";

export function useTranslation(): PluginI18n | null {
    return useContext(getGlobalI18nPluginContext());
}
