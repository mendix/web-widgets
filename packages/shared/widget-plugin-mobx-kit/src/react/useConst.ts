import { useRef } from "react";

export function useConst<T>(fn: T): T extends (...args: never) => infer R ? R : T {
    return (useRef<T | null>(null).current ??= typeof fn === "function" ? fn() : fn);
}
