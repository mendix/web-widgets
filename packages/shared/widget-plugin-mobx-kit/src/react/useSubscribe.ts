import { IReactionOptions, reaction } from "mobx";
import { useEffect, useState } from "react";

export function useSubscribe<Store extends object, T, FireImmediately extends boolean = false>(
    init: Store | (() => Store),
    selector: (store: Store) => T,
    options?: IReactionOptions<T, FireImmediately>
): [T, Store] {
    const [store] = useState(() => (typeof init === "function" ? init() : init));
    const [value, setValue] = useState(() => selector(store));

    function setup(): () => void {
        return reaction(() => selector(store), setValue, options);
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
    useEffect(setup, []);

    return [value, store];
}
