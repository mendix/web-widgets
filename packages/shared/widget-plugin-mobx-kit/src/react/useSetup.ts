import { useEffect } from "react";
import { useConst } from "./useConst";

export function useSetup<
    T extends {
        setup(): void | (() => void);
    }
>(fn: () => T): T {
    const obj = useConst(fn);
    useEffect(() => obj.setup(), [obj]);
    return obj;
}
