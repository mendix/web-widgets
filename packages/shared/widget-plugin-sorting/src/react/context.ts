import { useConst } from "@mendix/widget-plugin-mobx-kit/react/useConst";
import { generateUUID } from "@mendix/widget-plugin-platform/framework/generate-uuid";
import { Context, createContext, useContext, useEffect } from "react";
import { SortStoreHost } from "../stores/SortStoreHost";
import { Result, error, value } from "./result-meta";

export interface SortAPI {
    version: 1;
    host: SortStoreHost;
}

const SORT_PATH = "com.mendix.widgets.web.sortable.sortContext";

export function getGlobalSortContext(): Context<SortAPI | null> {
    const scope = window.top === window ? window : window.top;
    return ((scope as any)[SORT_PATH] ??= createContext<SortAPI | null>(null));
}

export function useSortAPI(): Result<SortAPI, Error> {
    const api = useContext(getGlobalSortContext());
    if (api === null) {
        return error(new Error("Error: widget is out of context. Please place the widget inside the Gallery header."));
    }
    return value(api);
}

export function useLockSortAPI(api: SortAPI): Result<SortAPI, Error> {
    const id = useLock(api);

    if (api.host.usedBy !== id) {
        return error(
            new Error(
                `Error: Sort API is already in use by another widget. Remove other sort widgets and refresh the page.`
            )
        );
    }

    return value(api);
}

function useLock({ host }: SortAPI): string {
    const [unlock, id] = useConst(() => {
        const id = `useLock@${generateUUID()}`;
        return [host.lock(id), id] as const;
    });

    useEffect(() => unlock, [unlock]);

    return id;
}
