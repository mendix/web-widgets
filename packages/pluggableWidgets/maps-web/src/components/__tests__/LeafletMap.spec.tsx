import "@testing-library/jest-dom";
import { render, RenderResult } from "@testing-library/react";
import { LeafletMap, LeafletProps } from "../LeafletMap";

describe("Leaflet maps", () => {
    const defaultProps: LeafletProps = {
        attributionControl: false,
        autoZoom: true,
        className: "",
        currentLocation: undefined,
        height: 75,
        heightUnit: "pixels",
        locations: [],
        mapProvider: "openStreet",
        mapsToken: "",
        optionDrag: true,
        optionScroll: true,
        optionZoomControl: true,
        showCurrentLocation: false,
        style: {},
        width: 50,
        widthUnit: "percentage",
        zoomLevel: 10
    };

    function renderLeafletMap(props: Partial<LeafletProps> = {}): RenderResult {
        return render(<LeafletMap {...defaultProps} {...props} />);
    }

    it("renders a map with right structure", () => {
        const { asFragment } = renderLeafletMap({ heightUnit: "percentageOfWidth", widthUnit: "pixels" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with pixels renders structure correctly", () => {
        const { asFragment } = renderLeafletMap({ heightUnit: "pixels", widthUnit: "pixels" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with percentage of width and height units renders the structure correctly", () => {
        const { asFragment } = renderLeafletMap({ heightUnit: "percentageOfWidth", widthUnit: "percentage" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with percentage of parent units renders the structure correctly", () => {
        const { asFragment } = renderLeafletMap({ heightUnit: "percentageOfParent", widthUnit: "percentage" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with HERE maps as provider", () => {
        const { asFragment } = renderLeafletMap({ mapProvider: "hereMaps" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with MapBox maps as provider", () => {
        const { asFragment } = renderLeafletMap({ mapProvider: "mapBox" });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with attribution", () => {
        const { asFragment } = renderLeafletMap({ attributionControl: true });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with markers", () => {
        const { asFragment } = renderLeafletMap({
            locations: [
                {
                    title: "Mendix HQ",
                    latitude: 51.906688,
                    longitude: 4.48837,
                    url: "image:url"
                },
                {
                    title: "Gementee Rotterdam",
                    latitude: 51.922823,
                    longitude: 4.479632,
                    url: "image:url"
                }
            ]
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders a map with current location", () => {
        const { asFragment } = renderLeafletMap({
            showCurrentLocation: true,
            currentLocation: {
                latitude: 51.906688,
                longitude: 4.48837,
                url: "image:url"
            }
        });
        expect(asFragment()).toMatchSnapshot();
    });
});
