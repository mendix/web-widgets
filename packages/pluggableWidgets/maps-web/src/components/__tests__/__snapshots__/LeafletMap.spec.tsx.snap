// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaflet maps renders a map with HERE maps as provider 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://2.base.maps.cit.api.here.com/maptile/2.1/maptile/newest/normal.day/10/524/338/256/png8"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with MapBox maps as provider 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://api.mapbox.com/styles/v1/mapbox/streets-v11/tiles/9/262/169"
                  style="width: 512px; height: 512px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with attribution 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                target="_blank"
                title="A JavaScript library for interactive maps"
              >
                <svg
                  aria-hidden="true"
                  class="leaflet-attribution-flag"
                  height="8"
                  viewBox="0 0 12 8"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h12v4H0z"
                    fill="#4C7BE1"
                  />
                  <path
                    d="M0 4h12v3H0z"
                    fill="#FFD500"
                  />
                  <path
                    d="M0 7h12v1H0z"
                    fill="#E0BC00"
                  />
                </svg>
                Leaflet
              </a>
               
              <span
                aria-hidden="true"
              >
                |
              </span>
               © 
              <a
                href="https://osm.org/copyright"
              >
                OpenStreetMap
              </a>
               contributors
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with current location 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <div
              class="leaflet-marker-icon custom-leaflet-map-icon-marker leaflet-zoom-animated"
              role="button"
              style="margin-left: -6px; margin-top: -6px; width: 12px; height: 12px; transform: translate3d(0px,0px,0); z-index: 0;"
              tabindex="0"
            >
              <img
                alt="map marker"
                class="custom-leaflet-map-icon-marker-icon"
                src="image:url"
              />
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with markers 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <div
              class="leaflet-marker-icon custom-leaflet-map-icon-marker leaflet-zoom-animated leaflet-interactive"
              role="button"
              style="margin-left: -6px; margin-top: -6px; width: 12px; height: 12px; transform: translate3d(0px,0px,0); z-index: 0;"
              tabindex="0"
              title="Mendix HQ"
            >
              <img
                alt="map marker"
                class="custom-leaflet-map-icon-marker-icon"
                src="image:url"
              />
            </div>
            <div
              class="leaflet-marker-icon custom-leaflet-map-icon-marker leaflet-zoom-animated leaflet-interactive"
              role="button"
              style="margin-left: -6px; margin-top: -6px; width: 12px; height: 12px; transform: translate3d(-6px,-19px,0); z-index: -19;"
              tabindex="0"
              title="Gementee Rotterdam"
            >
              <img
                alt="map marker"
                class="custom-leaflet-map-icon-marker-icon"
                src="image:url"
              />
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with percentage of parent units renders the structure correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: 75%;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with percentage of width and height units renders the structure correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50%; height: auto; padding-bottom: 37.5%;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with pixels renders structure correctly 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50px; height: 75px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Leaflet maps renders a map with right structure 1`] = `
<DocumentFragment>
  <div
    class="widget-maps"
    style="width: 50px; height: 37.5px;"
  >
    <div
      class="widget-leaflet-maps-wrapper"
    >
      <div
        class="widget-leaflet-maps leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab leaflet-touch-drag leaflet-touch-zoom"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="transform: translate3d(0px,0px,0);"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; transform: translate3d(0px,0px,0) scale(1);"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  src="https://b.tile.osm.org/10/524/338.png"
                  style="width: 256px; height: 256px; transform: translate3d(-196px,-172px,0);"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          />
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
          <div
            class="leaflet-proxy leaflet-zoom-animated"
            style="transform: translate3d(134340.33129244443px,86700.37048118314px,0) scale(512);"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          >
            <div
              class="leaflet-control-zoom leaflet-bar leaflet-control"
            >
              <a
                aria-disabled="false"
                aria-label="Zoom in"
                class="leaflet-control-zoom-in"
                href="#"
                role="button"
                title="Zoom in"
              >
                <span
                  aria-hidden="true"
                >
                  +
                </span>
              </a>
              <a
                aria-disabled="false"
                aria-label="Zoom out"
                class="leaflet-control-zoom-out"
                href="#"
                role="button"
                title="Zoom out"
              >
                <span
                  aria-hidden="true"
                >
                  −
                </span>
              </a>
            </div>
          </div>
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
